.github/workflows/dev-setup.yml

```diff
@@ -0,0 +1,90 @@
+name: Dev environment setup
```

Comment:
Do we need to run this workflow in any PRs? or only when we change `dev/dev-env-setup.sh`?